tests/js/realm-tests.js

```diff
@@ -1220,6 +1220,51 @@ module.exports = {
         }, 'The Realm file format must be allowed to be upgraded in order to proceed.');
     },
 
+    testSchemaUpdates: function() {
+        return new Promise((resolve, reject) => {
+            let calls = 0;
+            let realm1 = new Realm({_cache: false});
+            TestCase.assertEqual(realm1.schema.length, 0);  // empty schema
+            realm1.addListener('schema', (realm, event, schema) => {
+                calls++;
+
+                TestCase.assertEqual(event, 'schema');
+                TestCase.assertEqual(schema.length, 1);
+                TestCase.assertEqual(realm.schema.length, 1);
+                TestCase.assertEqual(schema[0].name, 'TestObject');
+                TestCase.assertEqual(realm1.schema.length, 1);
+                TestCase.assertEqual(realm.schema[0].name, 'TestObject');
+
+                if (calls == 2) {
+                    resolve();
+                }
+            });
+
+            realm1.addListener('change', (realm, event) => {
+                calls++;
+
+                TestCase.assertEqual(event, 'change');
+
+                if (calls == 2) {
+                    resolve();
+                }
+            });
+
+            const schema = [{
+                name: 'TestObject',
+                properties: {
+                    prop0: 'string',
+                }
+            }];
+            let realm2 = new Realm({ schema: schema, _cache: false });
+            TestCase.assertEqual(realm1.schema.length, 0); // not yet updated
+            TestCase.assertEqual(realm2.schema.length, 1);
+            realm2.write(() => {
```

Comment:
This also has the potential of giving us false positives in the case where there's a regression and the schema notifications only get triggered after a write transaction is committed to the table that has been added/changed.

Additionally, it would be nice to test that notifications are delivered both when a new class is added to the schema and when an existing class is updated with a new property.